content/blog/modern-text-features/index.Rmd

```diff
@@ -0,0 +1,370 @@
+This is not a jab at the provided graphic devices.
+ragg has handled text rendering in the exact same way up until now and the work that has gone into changing it has not been trivial.
+Technicalities aside, it is about time (overdue, actually) that the graphic system in R begins to support the display of non-Latin script and becomes more inclusive in terms of which languages can be used.
+It is thus with great joy that I can finally support it in ragg.
+
+### Right-to-Left scripts
+
+To start off we will look at a sample of different scripts that poses a problem due to their direction.
+
+```{r, eval=FALSE}
+arabic_text <- "هذا مكتوب باللغة العربية"
+hebrew_text <- "זה כתוב בעברית"
+sindhi_text <- "هي سنڌيءَ ۾ لکيو ويو آهي"
+
+p <- ggplot() + 
+  geom_text(
+    aes(x = 0, y = 3:1, label = c(arabic_text, hebrew_text, sindhi_text)), 
+    family = "Arial"
+  ) + 
+  expand_limits(y = c(0, 4))
+
+preview_devices(p, "rtl_example")
+```
+
+```{r, echo=FALSE, results='asis'}
+get_results("rtl_example")
+```
+
+Now, if you are not trained in reading any of the three languages above it may be hard to see what is right and what is wrong.
+You may, however, look at how the text in the code is rendered in the browser and compare that to the device rendering. If you do that you can see that the Hebrew script is rendered in the wrong direction for all the non-ragg devices (except Cairo on Linux).
+For the other scripts it is less obvious that the direction is wrong since the text glyphs are fundamentally different due to ligature substitutions.
+Still, by comparing to the browser rendering you can see that the same devices failing on the Hebrew script fail here as well.
+
+A point to make is that all of this "just works" in ragg.
+How is that?
+Shouldn't we have to indicate which kind of script we want to use?
+This is not necessary due to the genius of the Unicode standard which relates characters to specific scripts.
+The script, and by extension the layout, can thus simply be deduced from the provided string without needing to specify any additional information.
+
+The Cairo device on Linux handles this task well, as we have noted above.
+How come this works, but only on one OS (Cairo on macOS and Windows performs as bad as the other native devices)?
+Cairo is a fundamental library of many Linux distributions and is usually build on top of the Pango library which handles text layouting on Linux.
+It thus have access to OS level text rendering when used on Linux, but not on the other major platforms.
+The text rendering quality of the Cairo device on Linux is something that will be consistent through these examples and while it is very good, it is not fair to ask users to switch operating system to get modern font support.
+Even if you are already using Linux you'd have to accept that your output is not necessarily reproducible on other systems if you using the default device.
```

Comment:
```suggestion
Cairo is built in to most Linux distributions and is designed to work with Pango, the library that linux uses to layout text.
R's Cairo graphics device bundles Cairo on all platforms, but doesn't include Pango, due to the challenges of building it on other operating systems.
```

(I think the rest of the paragraph can be implied by the reader; I don't think a lot of people are going to rush to defend the Cairo device on linux)